extensions/wikia/ArticleVideo/scripts/article-video.js

```diff
@@ -0,0 +1,150 @@
+require(['wikia.window', 'wikia.onScroll', 'ooyalaVideo'], function (window, onScroll, OoyalaVideo) {
+	$(function () {
+		var $video = $('#article-video'),
+			$videoContainer = $video.find('.video-container'),
+			$videoThumbnail = $videoContainer.find('.video-thumbnail'),
+			$closeBtn = $videoContainer.find('.close'),
+			ooyalaVideoController,
+			ooyalaVideoElementId = 'ooyala-article-video',
+			$ooyalaVideo = $('#' + ooyalaVideoElementId),
+			videoCollapsed = false,
+			collapsingDisabled = false,
+			transitionEndEventName = getTransitionEndEventName();
+
+		function getTransitionEndEventName() {
+			var el = document.createElement('div'),
+				transitions = {
+					'transition': 'transitionend',
+					'OTransition': 'oTransitionEnd',
+					'MozTransition': 'transitionend',
+					'WebkitTransition': 'webkitTransitionEnd'
+				};
+
+			for (var t in transitions) {
+				if (el.style[t] !== undefined) {
+					return transitions[t];
+				}
+			}
+
+			return null;
+		}
+
+		function initVideo(onCreate) {
+			var ooyalaJsFile = window.wgArticleVideoData.jsUrl;
+			var ooyalaVideoId = window.wgArticleVideoData.videoId;
+
+			ooyalaVideoController = new OoyalaVideo(ooyalaVideoElementId, ooyalaJsFile, ooyalaVideoId, onCreate);
+		}
+
+		function uncollapseVideo() {
+			videoCollapsed = false;
+			$videoContainer.css({
+				'position': '',
+				'bottom': '',
+				'right': '',
+				'top': '',
+				'left': '',
+				'width': ''
+			});
+			$videoThumbnail.css('height', '');
+			$ooyalaVideo.css('height', '');
+			$video.removeClass('collapsed collapsed-ready');
+			if (ooyalaVideoController) {
+				updatePlayerControls(false);
+			}
+
+		}
+
+		function closeButtonClicked(event) {
+			event.stopPropagation();
+			if (ooyalaVideoController && ooyalaVideoController.player) {
+				ooyalaVideoController.player.pause();
+			}
+			uncollapseVideo();
+			collapsingDisabled = true;
+		}
+
+		function updatePlayerControls(waitForTransition) {
+			ooyalaVideoController.hideControls();
+			if (waitForTransition) {
+				$videoContainer.on(transitionEndEventName, function () {
+					ooyalaVideoController.sizeChanged();
+					ooyalaVideoController.showControls();
+				});
+			} else {
+				ooyalaVideoController.sizeChanged();
+				ooyalaVideoController.showControls();
+			}
+		}
+
+		function isVideoPlaying() {
+			if(ooyalaVideoController && ooyalaVideoController.player) {
+				return ooyalaVideoController.player.getState() === OO.STATE.PLAYING;
+			}
+		}
+
+		function isVideoInFullScreenMode() {
+			if(ooyalaVideoController && ooyalaVideoController.player) {
+				return ooyalaVideoController.player.getFullscreen();
+			}
+		}
+
+		function toggleCollapse() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)